main.tf

```diff
@@ -110,3 +111,28 @@ resource "aws_security_group_rule" "allow_all_outbound" {
 
   security_group_id = "${aws_security_group.main_db_access.id}"
 }
+
+resource "aws_iam_role" "rds_monitoring" {
+  name               = "${var.monitoring_role_name}"
```

Comment:
You can use `count = ${var.monitoring_interval == "0" ? 0 : 1}` on `aws_iam_role` and `aws_iam_role_policy_attachment` to not create resources if they are not needed

Reply:
Yes, I know, but I seem to recall that if you use the resource (as in the monitoring_role_arn) even if conditional, then it has to exist. I will check...